tutorials/spanruler_restaurant_reviews/scripts/attach_rules.py

```diff
@@ -0,0 +1,49 @@
+from enum import Enum
+from pathlib import Path
+
+import spacy
+import typer
+from srsly import read_jsonl
+from wasabi import msg
+
+Arg = typer.Argument
+Opt = typer.Option
+
+DEFAULT_PATTERNS_FILE = Path("patterns.jsonl")
+
+
+class Location(str, Enum):
+    before = "before"
+    after = "after"
+
+
+def attach_rules(
```

Comment:
I'm not sure if its a better approach, but I would prefer having a `config.cfg` for the full pipeline instead of running the `attach` script. 

This can be achieved I think by just implementing a factory

```python
def build_restaurant_ruler(
    nlp: Language,
    name: str,
    rules: Dict[str, Dict[str, str]],
    *,
    fields: Optional[List[str]] = None
) -> SpanRuler:
)
```
If you follow my suggestions with the rules being in a version function than in the config you can add something like:


```yaml
[components.restaurant_ruler]
factory = "RestaurantRuler.v1"
rules = {"@misc" : "restaurant_rules.v1"}
```

This way there is no script to stitch the `nlp` together, rather, we have registered factories for `SpanRuler` and its rules which are versioned and there is an additional config that has the combined component. I think this is more explicit and easier to version in the future. Maybe no so important, but I like this design.

Reply:
I would version the rules but not the factory. `.` is no longer supported in factory names either, factories are intended to be versioned by the installed module version, so you can continue to have `nlp.add_pipe("ner")` across multiple versions of spacy and not have to remember some specific version. But spacy v3.1.x always gets you the exact same component, while there might be slight changes across minor versions otherwise.

Reply:
Oh true, the rules are versioned and the factory takes the rules so there is no need to version the factory in any case. And what do you think about having the factory and adding the `span_ruler` in the config vs. in the `attach` script? To me the factory + config is the better more `spaCy`-like practice due to the explicitness.

Reply:
Sure will do that and remove the attach script to make everything based from the config